TripPlaner/src/main/java/com/singapore/TripPlaner/Service/CityService.java

```diff
@@ -26,4 +32,13 @@
 //        Function<City, String> onlyName = s->s.getName();
 //        return list.stream().map(onlyName).collect(Collectors.toList());
 //    }
-//}
\ No newline at end of file
+
+    public void addCity(City city) {
+//        Reader reader = new Reader();
+//        Persistent cityPersistent = (Persistent) reader.getList(City.class);
+//        List <City> cityList = (List<City>) cityPersistent;
+//        cityList.add(city);
+        Writer writer = new Writer();
+        writer.save((Persistent) city);
```

Comment:
tutaj nie można rzutować na Persistent. Wystarczy writer.save(city) - tak to on nie znajdzie w ogóle pliku
obiekt writer powinien być wstrzykiwany --> patrz reader w CityControlerze